content/2015.12.loadtesting.with.oauth.rst

```diff
@@ -0,0 +1,214 @@
+How to load-test a OAuth-protected HTTP API?
+############################################
+
+:lang: en
+:date: 2015-12-02
+:url: loadtesting-a-http-api-which-uses-oauth-for-authentication
+:summary: This article explains how to test the performance of an HTTP API when its access depends on an external OAuth service.
+
+
+Why would you need to run a loadtest?
+=====================================
+
+After deploying an HTTP API you might be wondering how many users or
+requests your deployment can take and what happens after it reaches
+that number.
+
+Knowing the maximum load your service can handle is a great thing to
+know!  It tells you when to start scaling up when your production
+server is close to reach that number of requests per seconds.
+
+
+So, what's so complicated about Oauth?
+======================================
+
+Loadtesting with `OAuth <https://en.wikipedia.org/wiki/OAuth>`_ means
+having an OAuth account, often creating and validating it and then
+getting `Bearer tokens <http://tools.ietf.org/html/rfc6750>`_ with the
+right scopes.
+
+Doing all this setup, will add a number of requests. Plus it will take
+time to wait for the email with the activation account link.
+
+This can take a tremendous amount of time in the scope of a loadtest
+which can slow down the all process and return inaccurate results.
+
+
+What is the right way to handle this?
+=====================================
+
+During a loadtest, make sure that your are only doing requests on the
+service you are loadtesting.
+
+If you are calling an identity provider or any other service during
+the loadtest phases (either setup, tear down or scenario phase) your
+loadtest performance will be impacted and will return inaccurate
+results.
+
+You should be able to write your loadtest with simple cURL commands
+toward your service.
+
+
+Configuration over code
+=======================
+
+If your service needs Bearer tokens to be loadtested, these Bearer
+tokens should be provided to your code loadtest and not be generated
+during the loadtest.
+
+This can be done by configuration, either within a file or with
+environment variables.
+
+
+Some code to configure your loadtest
+====================================
+
+With regards to the kind of environment you want to loadtest or
+smoketest on (dev, stage, production), you may want to configure your
```

Comment:
smoke test is introduced here but was not explained. It's vague what's the difference with load tests in this section